File: test_shopee_last_try/2a.py
import pandas as pd

df1 = pd.read_csv('Analisis excel/Youtube/5 soal/Customer.csv')
df1 = df1.drop_duplicates(('City','Customer ID'))
df1 = df1.drop_duplicates(['City','Customer ID'])
df1 = df1.drop_duplicates(subset =['City','Customer ID'])

df2 = pd.read_csv('Analisis excel/Youtube/5 soal/Orders.csv')



df = pd.merge(df1, df2, on='Customer ID', how='inner')
print(df.head())


hasil = df.groupby('City')['Order ID'].count().reset_index(name='jumlah pembelian')
print(hasil)

File: test_shopee_last_try/2a.r
library(dplyr)

customer <- read.csv("Youtube/5 soal/Customer.csv")
customer <- customer %>% distinct(City, Customer.ID, .keep_all = TRUE)


print(head(customer))

order <- read.csv("Youtube/5 soal/Orders.csv")
print(head(order))
cat('\n')


# Inner join customer and order data frames
hasil1 <- inner_join(customer, order, by = "Customer.ID", relationship ="many-to-many")
print(head(hasil1))

cat("\n\n")

# Alternative approach using merge
hasil2 <- merge(customer, order, by = "Customer.ID", all = FALSE)
print(head(hasil2))

hasil_akhir <- hasil1 %>% group_by(City) %>% summarise(jumlah = n()) 
print(hasil_akhir)

hasil_akhir <- hasil2 %>% group_by(City) %>% summarise(jumlah = n()) 
print(hasil_akhir)